numbat/src/ffi/lists.rs

```diff
@@ -23,11 +25,8 @@ pub fn head(mut args: Args) -> Result<Value> {
 pub fn tail(mut args: Args) -> Result<Value> {
     let mut list = list_arg!(args);
 
-    if list.pop_front().is_some() {
-        return_list!(list)
-    } else {
-        Err(RuntimeError::EmptyList)
-    }
+    list.advance_view()?;
```

Comment:
Minor "advance view" is sort of a implementation detail. Maybe we can call that function `tail` or `into_tail` or similar?